doc/user/content/sql/create-source/kafka.md

```diff
@@ -0,0 +1,372 @@
+
+If you need to limit the amount of data maintained as state after source creation, consider using [temporal filters](/guides/temporal-filters/) instead.
+
+#### `WITH` options
+
+Field               | Value | Description
+--------------------|-------|--------------------
+`start_offset`      | `int` | Read partitions from the specified offset. You cannot update the offsets once a source has been created; you will need to recreate the source. Offset values must be zero or positive integers, and the source must use either `ENVELOPE NONE` or `ENVELOPE DEBEZIUM UPSERT`.
+`kafka_time_offset` | `int` | Use the specified value to set `start_offset` based on the Kafka timestamp. Negative values will be interpreted as relative to the current system time in milliseconds (e.g. `-1000` means 1000 ms ago). The offset for each partition will be the earliest offset whose timestamp is greater than or equal to the given timestamp in the corresponding partition. If no such offset exists for a partition, the partition's end offset will be used. **This option is not currently supported for [Redpanda](/third-party/redpanda).**
+
+
+## Authentication
+
+### SSL
+
+To connect to a Kafka broker that requires [SSL authentication](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html), use the provided [`WITH` options](#ssl-with-options).
+
+```sql
+CREATE MATERIALIZED SOURCE kafka_ssl
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'top-secret' WITH (
+      security_protocol = 'SSL',
+      ssl_key_location = '/secrets/materialized.key',
+      ssl_certificate_location = '/secrets/materialized.crt',
+      ssl_ca_location = '/secrets/ca.crt',
+      ssl_key_password = 'mzmzmz'
+  )
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'https://localhost:8081' WITH (
+      ssl_key_location = '/secrets/materialized.key',
+      ssl_certificate_location = '/secrets/materialized.crt',
+      ssl_ca_location = '/secrets/ca.crt',
+  );
+```
+
+Note that:
+
+- To connect to a Confluent Schema Registry using SSL, it must use the same certificate authority (CA) as the Kafka broker.
+
+#### SSL `WITH` options
+
+Field                      | Value  | Description
+---------------------------|--------|---------------------------
+`security_protocol`        | `text` | Use `ssl` to connect to the Kafka cluster.
+`ssl_certificate_location` | `text` | The absolute path to your SSL certificate. Required for SSL client authentication.
+`ssl_key_location`         | `text` | The absolute path to your SSL certificate's key. Required for SSL client authentication.
+`ssl_key_password`         | `text` | Your SSL key's password, if any.
+`ssl_ca_location`          | `text` | The absolute path to the certificate authority (CA) certificate. Used for both SSL client and server authentication. If unspecified, uses the system's default CA certificates.
+
+#### Confluent Schema Registry SSL `WITH` options
```

Comment:
This is the one part that's a bit weird to me in terms of the information hierarchy. All the other `WITH` options refer to the Kafka _connector's_ with options. This one refers to the `format_spec`'s `WITH` options. Maybe that's just fine. It's super weird how tightly coupled the `USING CONFLUENT SCHEMA REGISTRY` clause is to Kafka sources/sinks anyway.